kasmvnc/run.sh

```diff
@@ -14,143 +15,137 @@ check_installed() {
     exit 1
   fi
-}
 
-# Function to install kasmvncserver for debian-based distros
-install_deb() {
-  local url=$1
-  download_file $url /tmp/kasmvncserver.deb
-  sudo apt-get update
-  DEBIAN_FRONTEND=noninteractive sudo apt-get install --yes -qq --no-install-recommends --no-install-suggests /tmp/kasmvncserver.deb
-  sudo adduser $USER ssl-cert
-  rm /tmp/kasmvncserver.deb
+  $download_tool "$url" > "$output" || {
+    echo "ERROR: Failed to download $url"
+    exit 1
+  }
 }
 
-# Function to install kasmvncserver for Oracle 8
-install_rpm_oracle8() {
-  local url=$1
-  download_file $url /tmp/kasmvncserver.rpm
-  sudo dnf config-manager --set-enabled ol8_codeready_builder
-  sudo dnf install oracle-epel-release-el8 -y
-  sudo dnf localinstall /tmp/kasmvncserver.rpm -y
-  sudo usermod -aG kasmvnc-cert $USER
-  rm /tmp/kasmvncserver.rpm
+# Add user to group using available commands
+add_user_to_group() {
+  local user="$1"
+  local group="$2"
+
+  if command -v usermod &> /dev/null; then
+    sudo usermod -aG "$group" "$user"
+  elif command -v adduser &> /dev/null; then
+    sudo adduser "$user" "$group"
+  else
+    echo "ERROR: At least one of 'adduser'(Debian) 'usermod'(RHEL) is required"
+    exit 1
+  fi
 }
 
-# Function to install kasmvncserver for CentOS 7
-install_rpm_centos7() {
+# Function to install kasmvncserver for debian-based distros
+install_deb() {
   local url=$1
-  download_file $url /tmp/kasmvncserver.rpm
-  sudo yum install epel-release -y
-  sudo yum install /tmp/kasmvncserver.rpm -y
-  sudo usermod -aG kasmvnc-cert $USER
-  rm /tmp/kasmvncserver.rpm
+  download_file "$url" /tmp/kasmvncserver.deb
+  # Define the directory to check
+  CACHE_DIR="/var/lib/apt/lists/partial"
+  # Check if the directory exists and was modified in the last 60 minutes
+  if [ ! -d "$CACHE_DIR" ] || ! find "$CACHE_DIR" -mmin -60 -print -quit &> /dev/null; then
+    echo "Stale Package Cache, updating..."
+    # Update package cache with a 300-second timeout for dpkg lock
+    sudo apt-get -o DPkg::Lock::Timeout=300 -qq update
+  fi
+  DEBIAN_FRONTEND=noninteractive sudo apt-get -o DPkg::Lock::Timeout=300 install --yes -qq --no-install-recommends --no-install-suggests /tmp/kasmvncserver.deb
+  add_user_to_group "$USER" ssl-cert
+  rm /tmp/kasmvncserver.deb
 }
 
 # Function to install kasmvncserver for rpm-based distros
 install_rpm() {
   local url=$1
-  download_file $url /tmp/kasmvncserver.rpm
+  download_file "$url" /tmp/kasmvncserver.rpm
   sudo rpm -i /tmp/kasmvncserver.rpm
   rm /tmp/kasmvncserver.rpm
 }
 
 # Function to install kasmvncserver for Alpine Linux
 install_alpine() {
   local url=$1
-  download_file $url /tmp/kasmvncserver.tgz
+  download_file "$url" /tmp/kasmvncserver.tgz
   tar -xzf /tmp/kasmvncserver.tgz -C /usr/local/bin/
   rm /tmp/kasmvncserver.tgz
 }
 
+# Check for sudo (required)
+if ! command -v sudo &> /dev/null; then
+  echo "ERROR: Required command 'sudo' not found"
+  exit 1
+fi
```

Comment:
I am testing this with https://kasmweb.com/docs/latest/guide/custom_images.html images, which do not have sudo access for the `kasm-user` container user but have kasmvnc pre-installed.

A WIP can be seen in the [`support-kasm-images`](https://github.com/coder/modules/tree/atif/support-kasm-images) branch.

What I did was to check if `vncserver` command is already available and then skip the installation and directly move to starting the vncserver.